app_test.go

```diff
@@ -113,7 +113,30 @@ func TestNewApp(t *testing.T) {
 			[]string{"Provided", "Provided", "Provided", "Provided", "LoggerInitialized", "Started"},
 			spy.EventTypes())
 
-		assert.Contains(t, spy.Events()[0].(*fxevent.Provided).OutputTypeNames, "struct {}")
+		// Lifecycle, Shutdowner, and DotGraph are indices 0, 1, and 2.
```

Comment:
Want to assert this too? I get that this feels a bit change-detector ish but it already is that way as this test is structured. We may want to consider refactoring these tests in the future to reduce its change-detector ishness.